core/vm/cip20.go

```diff
@@ -0,0 +1,207 @@
+package vm
+
+import (
+	"errors"
+
+	"crypto/sha512"
+
+	"github.com/ethereum/go-ethereum/crypto/blake2s"
+	"github.com/ethereum/go-ethereum/params"
+	"golang.org/x/crypto/sha3"
+)
+
+const (
```

Comment:
Word is quite overloaded in the current context. (e.g. it could mean machine word, EVM word, or the hash input unit here) Could you clarify what is meant here in a comment and note that the units are bytes?
```suggestion
// Token unit of length, in bytes, used for calculating gas with respect to each support hash function.
const (
```

Reply:
Flagging that if we want to change metering, [CIP-25](https://github.com/celo-org/celo-blockchain/pull/1256) uses a similar metering method/namings

Reply:
yeah, the overloading of word created this ugly patch in the spec. Initially I assumed the EVM authors meant hash function word when they actually meant EVM word. If we're going to redo the specification, I would suggest pricing everything by 32-byte words, but leaving prices unchanged

Reply:
That would make sense. It would give a consistent and way to talk about the gas pricing. By "leave the prices unchanged" do you mean have the same cost/word with reduced word size (i.e. double the cost per byte for SHA3-512) or do you mean to keep the same cost/byte? I would be most supportive of either making the cost/byte match of SHA3-512 match SHA3-256, or be twice as large, because it seems to require twice as many sponge function invocations on average.

Reply:
my opinion is that all hash functions are severely overpriced, so I am biased towards leaving it the same as sha3-256. Because this is a precompile we have the additional CALL gas overhead that prevents abuse as well.

I have updated the PR to standardize the usage of "word" as an EVM word